Geek-Collectors-Network/geek-collectors-network
# 74 extract formcontrol component to promote code reusability

## Summary
Attempted to break down the forms, `LoginForm` and `RegistrationForm`, into smaller pieces.

- Exported login controls, consisting of `Remember me?` and `Forgot password`, into a new file called `Login Controls`
- Exported validation schemas into a separate `ValidationSchemas` file
- Created a `TextInput` component. This uses `useFormat()` hook, which receives state, validation, and submission from `<Formik>`. We deconstruct the information received into two fields, `field`, and `meta`, which we spread across the appropriate pieces in `TextInput`.
- Use this `TextInput` field in `RegistrationForm` and `LoginForm`, respectively. 

## Testing

1. Navigate to the landing page using Deploy Preview (below)
2. Navigate to the `Login`
3. Click `Submit` and confirm that the email and password fields outline in red and display error messages
4. Fill in the email, don't fill in password, click `Submit`, and confirm the password field goes red and displays an error
5. Fill in the password, don't fill in email, click `Submit`, and confirm the email field goes red and displays an error
6. Fill in both fields with valid inputs and click the `Submit`; confirm, in Google Developer Tools, that an object is created containing both the entered email and password.
